eth/protocols/eth/peer.go

```diff
@@ -465,3 +424,35 @@ func (p *Peer) RequestTxs(hashes []common.Hash) error {
 		GetPooledTransactionsPacket: hashes,
 	})
 }
+
```

Comment:
`leakySet` is not the best name, it's essentially a `cache`. Could you call it `knownCache` or something similar?